src/layouts/Layout.astro

```diff
@@ -14,6 +14,130 @@ const { title } = Astro.props;
 		<meta charset="UTF-8" />
 		<meta name="viewport" content="width=device-width" />
 		<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+				<script lang="js">
+			/**
+			 * @param tryOnce
+			 * @returns {Promise<void|ServiceWorker|*>}
+			 *
+			 * @see https://stackoverflow.com/a/67612740
+			 */
+			async function registerServiceWorker(tryOnce = false) {
+				if (!('serviceWorker' in navigator)) throw new Error('serviceWorker not supported');
+
+				const url = (new URL(`/serviceWorker.js`, location)).toString();
+				//console.info('Registering worker');
+				const registration = await navigator.serviceWorker.register(url, {
+					scope: '/',
+				});
+
+				const registeredWorker = registration.active || registration.waiting || registration.installing;
+				//console.info('Registered worker:', registeredWorker);
+				if (registeredWorker?.scriptURL != url) {
+					console.log('[ServiceWorker] Old URL:', registeredWorker?.scriptURL || 'none', 'updating to:', url);
+					await registration.update();
+					//console.info('Updated worker');
+				}
+
+				//console.info('Waiting for ready worker');
+				let serviceReg = await navigator.serviceWorker.ready;
+				//console.info('Ready registration:', serviceReg);
+
+				if (!navigator.serviceWorker.controller) {
+					//console.info('Worker isn’t controlling, re-register');
+					try {
+						const reg = await navigator.serviceWorker.getRegistration('/');
+						//console.info('Unregistering worker');
+						await reg.unregister();
+						//console.info('Successfully unregistered, trying registration again');
+						return registerServiceWorker();
+					} catch (err) {
+						console.error(`ServiceWorker failed to re-register after hard-refresh, reloading the page!`, err);
+						return location.reload();
+					}
+				}
+
+				let serviceWorker = serviceReg.active || serviceReg.waiting || serviceReg.installing;
+				if (!serviceWorker) {
+					//console.info('No worker on registration, getting registration again');
+					serviceReg = await navigator.serviceWorker.getRegistration('/');
+					serviceWorker = serviceReg.active || serviceReg.waiting || serviceReg.installing;
+				}
+
+				if (!serviceWorker) {
+					//console.info('No worker on registration, waiting 50ms');
+					await sleep(50); // adjustable or skippable, have a play around
+				}
```

Comment:
What can happen if we don't wait 50 ms?
Can't we use await or so, to make sure serviceWorker is set on line 63, instead of using this high tech sleep method?